syncano/models/fields.py

```diff
@@ -0,0 +1,270 @@
+
+
+class StringField(WritableField):
+
+    def to_python(self, value):
+        if isinstance(value, six.string_types) or value is None:
+            return value
+        return six.u(value)
+
+
+class IntegerField(WritableField):
+
+    def to_python(self, value):
+        if value is None:
+            return value
+        try:
+            return int(value)
+        except (TypeError, ValueError):
+            raise self.VaidationError('Invalid value.')
+
+
+class FloatField(WritableField):
+
+    def to_python(self, value):
+        if value is None:
+            return value
+        try:
+            return float(value)
+        except (TypeError, ValueError):
+            raise self.VaidationError('Invalid value.')
+
+
+class BooleanField(WritableField):
+
+    def to_python(self, value):
+        if value in (True, False):
+            return bool(value)
+
+        if value in ('t', 'True', '1'):
+            return True
+
+        if value in ('f', 'False', '0'):
+            return False
+
+        raise self.VaidationError('Invalid value.')
+
+
+class SlugField(StringField):
+    regex = re.compile(r'^[-a-zA-Z0-9_]+$')
+
+    def validate(self, value, model_instance):
+        super(SlugField, self).validate(value, model_instance)
+        if not bool(self.regex.search(value)):
+            raise self.VaidationError('Invalid value.')
+        return value
+
+
+class EmailField(StringField):
+    regex = re.compile(r'(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)')
+
+    def validate(self, value, model_instance):
+        super(EmailField, self).validate(value, model_instance)
+
+        if not value or '@' not in value:
+            raise self.VaidationError('Enter a valid email address.')
+
+        if not bool(self.regex.match(value)):
+            raise self.VaidationError('Enter a valid email address.')
+
+
+class ChoiceField(WritableField):
+
+    def __init__(self, *args, **kwargs):
+        self.choices = kwargs.pop('choices', [])
+        self.allowed_values = [choice['value'] for choice in self.choices]
+        super(ChoiceField, self).__init__(*args, **kwargs)
+
+    def validate(self, value, model_instance):
+        super(ChoiceField, self).validate(value, model_instance)
+        if self.choices and value not in self.allowed_values:
+            raise self.VaidationError('Invalid choice.')
+
+
+class DateField(WritableField):
+    date_regex = re = re.compile(
+        r'(?P<year>\d{4})-(?P<month>\d{1,2})-(?P<day>\d{1,2})$'
+    )
+
+    def to_python(self, value):
+        if value is None:
+            return value
+
+        if isinstance(value, datetime):
+            return value.date()
+
+        if isinstance(value, date):
+            return value
+
+        try:
+            parsed = self.parse_date(value)
+            if parsed is not None:
+                return parsed
+        except ValueError:
+            pass
+
+        raise self.VaidationError('Invalid date.')
```

Comment:
Add information about date format here. It's not obvious for everyone. Especially Europeans vs Americans ;).